# Healthcare Marketing Analytics Project
## Determining the impact of R&D Ration on Healthcare Firms' Value

In [2]:
import warnings
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
warnings.simplefilter("once")

### Data Sourcing
This dataset was provided by Bocconi University, Marketing Analytics department. Here is the description of the data columns in the dataset:
- Firm ID (variable name: id)
- Year (variable name: year)
- Firm Total Assets (unit: million $) (variable name: assets)
- Long-term Debt (unit: million $) (variable name: debt)
- Earnings (unit: million $): This variable is EBITDA (earnings before interest, taxes, depreciation,
and amortization). (variable name: earnings)
- Cash Flows (unit: million $): This is Net Cash Flow from Operating Activity. However, please
conceptually consider this variable a proxy for profit based on only cash as some academic
studies do. (variable name: cashflow)
- Sales (unit: million $) (variable name: sales)
- Advertising Expenditure (unit: million $): Conceptually, you can consider this variable as a firm’s
brand building effort. (variable name: ad)
- R&D Expenditure (unit: million $): Conceptually, you can consider this variable as a firm’s
innovation effort. (variable name: rd)
- Marketing, Selling, and Operating Expenditure (unit: million $): Conceptually, you can consider
this variable as a firm’s selling and marketing effort (e.g., sale forces, customer management)
except advertising. Note that this variable does not include advertising expenditure and R&D
expenditure. (variable name: mkt)
- Product Market Threat: Consider this variable as competitive threats that a firm goes through in
its product markets. Higher values indicate higher levels of competitive threats. (variable name:
threat)
- 2001 Recession (dummy): 1 if the observation is in the year of 2001 recession and 0 otherwise.
(variable name: rec1)
- 2008 Recession (dummy): 1 if the observation is in the year of 2008 recession and 0 otherwise.
(variable name: rec2)
- Firm Value (i.e., Tobin’s Q) (variable name: fv)
- Market Valuation (unit: million $): Market value of a firm (variable name: mv)
- Business sector where a firm operates (indicator) (variable name: sector)
    - 1: Manufacturing (Consumer Durables and Non-Durables) e.g., Food, Tobacco, Textile, Apparel, Toys, Cars, TVs, Furniture, Appliances
    - 2: Manufacturing (Non-Consumer Goods) e.g., Machinery, Office Furniture, Paper, Computers, Electronic Equipment
    - 3: Wholesale
    - 4: Retailing
    - 5: Healthcare e.g., healthcare, medical equipment, drugs
    - 6: Information Technology e.g., software publishers, telecommunications, web search portals
    - 7: Service Sector (Technical Services, Professional and Administrative Services, and Education Services)
    - 8: Entertainment, Accommodation, and Full Service Restaurants

In [84]:
all_data = pd.read_csv('marketing_insight.csv')
all_data.head()

id  year  assets    debt  earnings  cashflow   sales          mv     ad  \
0    42  1993   4.103   0.000    -1.748    -2.175   3.610   63.674000  0.074   
1  2607  1993  18.897   0.716    -2.458    -2.016  26.274   10.802283  0.748   
2  1204  1993  37.476  23.735     4.359    -0.087  63.938   29.856251  0.759   
3   899  1993   7.882   0.011    -4.933    -4.958   5.411   58.403999  0.775   
4  1746  1993  24.496   0.000     7.723    -5.843  43.030  208.065000  0.719   

      rd   threat         fv  rec1  rec2    mkt  sector  
0  0.305   1.7482  14.836461     0     0  2.042       5  
1  1.310  18.0228   0.083362     0     0  5.405       1  
2  0.542   1.0184   1.225564     0     0  9.997       3  
3  2.059   9.1024   6.930602     0     0  6.089       5  
4  0.372   1.0658   7.952849     0     0  9.994       5

In [85]:
all_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22723 entries, 0 to 22722
Data columns (total 16 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   id        22723 non-null  int64  
 1   year      22723 non-null  int64  
 2   assets    22723 non-null  float64
 3   debt      22723 non-null  float64
 4   earnings  22723 non-null  float64
 5   cashflow  22723 non-null  float64
 6   sales     22723 non-null  float64
 7   mv        22723 non-null  float64
 8   ad        22723 non-null  float64
 9   rd        22723 non-null  float64
 10  threat    22723 non-null  float64
 11  fv        22723 non-null  float64
 12  rec1      22723 non-null  int64  
 13  rec2      22723 non-null  int64  
 14  mkt       22723 non-null  float64
 15  sector    22723 non-null  int64  
dtypes: float64(11), int64(5)
memory usage: 2.8 MB


#### Filtering out the Healthcare Sector
This project will be focused on the Healthcare sector, therefore, we'll filter out other sectors to remain with **sector 5** – ***Healthcare***.

In [86]:
data = all_data[all_data['sector']==5] #Filtering out all sectors to remain with healthcare data
data = data.drop(['sector'], axis=1) #Delete the sector column as it is no longer relevant
data.head()

id  year  assets   debt  earnings  cashflow   sales          mv     ad  \
0     42  1993   4.103  0.000    -1.748    -2.175   3.610   63.674000  0.074   
3    899  1993   7.882  0.011    -4.933    -4.958   5.411   58.403999  0.775   
4   1746  1993  24.496  0.000     7.723    -5.843  43.030  208.065000  0.719   
10  1312  1993  10.248  0.000   -15.562    -9.349   0.000   23.978001  0.000   
19  1686  1993   6.007  0.929     1.055     0.483   6.980   19.942678  0.086   

       rd  threat         fv  rec1  rec2     mkt  
0   0.305  1.7482  14.836461     0     0   2.042  
3   2.059  9.1024   6.930602     0     0   6.089  
4   0.372  1.0658   7.952849     0     0   9.994  
10  4.075  5.0967   2.646077     0     0  11.487  
19  0.197  1.0549   3.027747     0     0   1.558

In [87]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3292 entries, 0 to 22569
Data columns (total 15 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   id        3292 non-null   int64  
 1   year      3292 non-null   int64  
 2   assets    3292 non-null   float64
 3   debt      3292 non-null   float64
 4   earnings  3292 non-null   float64
 5   cashflow  3292 non-null   float64
 6   sales     3292 non-null   float64
 7   mv        3292 non-null   float64
 8   ad        3292 non-null   float64
 9   rd        3292 non-null   float64
 10  threat    3292 non-null   float64
 11  fv        3292 non-null   float64
 12  rec1      3292 non-null   int64  
 13  rec2      3292 non-null   int64  
 14  mkt       3292 non-null   float64
dtypes: float64(11), int64(4)
memory usage: 411.5 KB


In [88]:
data.describe()

id         year         assets          debt      earnings  \
count  3292.000000  3292.000000    3292.000000   3292.000000   3292.000000   
mean   1724.311665  2007.315614    3429.609149    775.250014    563.978191   
std    1058.535967     7.179453   16334.438091   3832.730215   2663.596820   
min       1.000000  1993.000000       0.404000      0.000000   -666.676000   
25%     774.000000  2002.000000      34.815250      0.000000     -5.244250   
50%    1584.000000  2007.000000     109.109000      1.957500      4.940000   
75%    2629.000000  2013.000000     495.226000     65.151500     58.128250   
max    3523.000000  2019.000000  212949.000000  63226.000000  28795.000000   

           cashflow         sales             mv           ad            rd  \
count   3292.000000   3292.000000    3292.000000  3292.000000   3292.000000   
mean     418.321345   1706.000846    6780.085848    71.307075    255.238051   
std     2038.262931   7211.308404   29286.544040   354.174284   1154.296713   
min     -635.442000      0.000000       0.809150     0.000000      0.000000   
25%       -4.947750     26.508250      52.787131     0.288000      1.748750   
50%        3.104500     84.768000     207.849600     1.103000      6.816000   
75%       39.787500    354.252500    1014.843800     6.500000     30.600500   
max    23416.000000  82059.000000  384003.780000  4000.000000  12245.000000   

            threat           fv         rec1         rec2           mkt  
count  3292.000000  3292.000000  3292.000000  3292.000000   3292.000000  
mean      8.154948     2.360336     0.034933     0.045261    408.670110  
std      11.103680     2.746864     0.183638     0.207908   1715.608140  
min       1.000000    -0.586277     0.000000     0.000000  -1895.591100  
25%       1.294500     0.861938     0.000000     0.000000     12.847250  
50%       3.197450     1.604317     0.000000     0.000000     34.998501  
75%      10.694725     2.901638     0.000000     0.000000    119.889250  
max      85.363701    45.089115     1.000000     1.000000  19088.000000

#### Generating Additional Variables
Here are varaiables we are generating:
- ***R&D Ratio (rd_ratio):*** R&D Expenditure/Sales 
- ***Strategic Emphasis (se):*** (Advertising Expenditure - R&D Expenditure)/Firm Total Assets
- ***Advertising Ratio (ad_ratio):*** Advertising Expenditure/Sales

**Note:** We'll drop instances where Sales and Assets is equal to 0 first, as this will output infinite values for RD_RATIO, AD_RATIO and SE. However, we are positive from the data summary that there are no firms with Assets as 0 (min value was 0.404000)

In [89]:
data = data[data['sales'] > 0] #Drop Instances where Sales is equal to 0
data = data[data['assets'] > 0] #Drop Instances where Assets is equal to 0
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3279 entries, 0 to 22569
Data columns (total 15 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   id        3279 non-null   int64  
 1   year      3279 non-null   int64  
 2   assets    3279 non-null   float64
 3   debt      3279 non-null   float64
 4   earnings  3279 non-null   float64
 5   cashflow  3279 non-null   float64
 6   sales     3279 non-null   float64
 7   mv        3279 non-null   float64
 8   ad        3279 non-null   float64
 9   rd        3279 non-null   float64
 10  threat    3279 non-null   float64
 11  fv        3279 non-null   float64
 12  rec1      3279 non-null   int64  
 13  rec2      3279 non-null   int64  
 14  mkt       3279 non-null   float64
dtypes: float64(11), int64(4)
memory usage: 409.9 KB


In [90]:
#Generate R&D Ratio
data['rd_ratio'] = data['rd']/data['sales']
data.head()

id  year  assets   debt  earnings  cashflow   sales          mv     ad  \
0     42  1993   4.103  0.000    -1.748    -2.175   3.610   63.674000  0.074   
3    899  1993   7.882  0.011    -4.933    -4.958   5.411   58.403999  0.775   
4   1746  1993  24.496  0.000     7.723    -5.843  43.030  208.065000  0.719   
19  1686  1993   6.007  0.929     1.055     0.483   6.980   19.942678  0.086   
23   214  1993  17.609  7.213    -7.784    -9.292   8.150  132.936000  0.600   

       rd  threat         fv  rec1  rec2    mkt  rd_ratio  
0   0.305  1.7482  14.836461     0     0  2.042  0.084488  
3   2.059  9.1024   6.930602     0     0  6.089  0.380521  
4   0.372  1.0658   7.952849     0     0  9.994  0.008645  
19  0.197  1.0549   3.027747     0     0  1.558  0.028223  
23  5.291  1.0538   7.814527     0     0  5.462  0.649202

In [91]:
#Generate Strategic Emphasis
data['se'] = (data['ad'] - data['rd'])/data['assets']
data.head()

id  year  assets   debt  earnings  cashflow   sales          mv     ad  \
0     42  1993   4.103  0.000    -1.748    -2.175   3.610   63.674000  0.074   
3    899  1993   7.882  0.011    -4.933    -4.958   5.411   58.403999  0.775   
4   1746  1993  24.496  0.000     7.723    -5.843  43.030  208.065000  0.719   
19  1686  1993   6.007  0.929     1.055     0.483   6.980   19.942678  0.086   
23   214  1993  17.609  7.213    -7.784    -9.292   8.150  132.936000  0.600   

       rd  threat         fv  rec1  rec2    mkt  rd_ratio        se  
0   0.305  1.7482  14.836461     0     0  2.042  0.084488 -0.056300  
3   2.059  9.1024   6.930602     0     0  6.089  0.380521 -0.162903  
4   0.372  1.0658   7.952849     0     0  9.994  0.008645  0.014166  
19  0.197  1.0549   3.027747     0     0  1.558  0.028223 -0.018478  
23  5.291  1.0538   7.814527     0     0  5.462  0.649202 -0.266398

In [92]:
#Generate Advertising Ratio
data['ad_ratio'] = data['ad']/data['sales']
data.head()

id  year  assets   debt  earnings  cashflow   sales          mv     ad  \
0     42  1993   4.103  0.000    -1.748    -2.175   3.610   63.674000  0.074   
3    899  1993   7.882  0.011    -4.933    -4.958   5.411   58.403999  0.775   
4   1746  1993  24.496  0.000     7.723    -5.843  43.030  208.065000  0.719   
19  1686  1993   6.007  0.929     1.055     0.483   6.980   19.942678  0.086   
23   214  1993  17.609  7.213    -7.784    -9.292   8.150  132.936000  0.600   

       rd  threat         fv  rec1  rec2    mkt  rd_ratio        se  ad_ratio  
0   0.305  1.7482  14.836461     0     0  2.042  0.084488 -0.056300  0.020499  
3   2.059  9.1024   6.930602     0     0  6.089  0.380521 -0.162903  0.143227  
4   0.372  1.0658   7.952849     0     0  9.994  0.008645  0.014166  0.016709  
19  0.197  1.0549   3.027747     0     0  1.558  0.028223 -0.018478  0.012321  
23  5.291  1.0538   7.814527     0     0  5.462  0.649202 -0.266398  0.073620

In [93]:
data.describe()

id        year         assets          debt      earnings  \
count  3279.000000  3279.00000    3279.000000   3279.000000   3279.000000   
mean   1722.858188  2007.31900    3443.012375    778.293995    566.307734   
std    1058.002425     7.17523   16365.404999   3840.016879   2668.615472   
min       1.000000  1993.00000       0.404000      0.000000   -666.676000   
25%     774.000000  2002.00000      35.078000      0.000000     -5.128500   
50%    1584.000000  2007.00000     109.719000      2.023000      5.005000   
75%    2629.000000  2013.00000     497.688500     66.958000     59.712000   
max    3523.000000  2019.00000  212949.000000  63226.000000  28795.000000   

           cashflow         sales             mv           ad            rd  \
count   3279.000000   3279.000000    3279.000000  3279.000000   3279.000000   
mean     420.058745   1712.764497    6806.196857    71.587884    256.210915   
std     2042.113092   7224.791755   29341.612756   354.847742   1156.479480   
min     -635.442000      0.008000       0.809150     0.000000      0.000000   
25%       -4.809000     26.847500      53.140995     0.288500      1.748500   
50%        3.170000     85.591000     208.300980     1.122000      6.825000   
75%       40.894500    355.725000    1017.885350     6.600000     30.664500   
max    23416.000000  82059.000000  384003.780000  4000.000000  12245.000000   

            threat           fv         rec1         rec2           mkt  \
count  3279.000000  3279.000000  3279.000000  3279.000000   3279.000000   
mean      8.112043     2.345257     0.034767     0.045441    410.237707   
std      11.059323     2.725278     0.183216     0.208300   1718.825467   
min       1.000000    -0.586277     0.000000     0.000000  -1895.591100   
25%       1.291950     0.860777     0.000000     0.000000     12.899500   
50%       3.159500     1.597478     0.000000     0.000000     35.310001   
75%      10.643300     2.898828     0.000000     0.000000    120.445500   
max      85.363701    45.089115     1.000000     1.000000  19088.000000   

          rd_ratio           se     ad_ratio  
count  3279.000000  3279.000000  3279.000000  
mean      0.472146    -0.072976     0.064353  
std       7.358644     0.149301     0.351981  
min       0.000000    -3.844961     0.000000  
25%       0.040958    -0.108612     0.004649  
50%       0.090683    -0.051508     0.014198  
75%       0.203036    -0.013301     0.042577  
max     395.785707     1.169534    14.142857

In [94]:
data['mkt_growth'] = ""
data.head()

id  year  assets   debt  earnings  cashflow   sales          mv     ad  \
0     42  1993   4.103  0.000    -1.748    -2.175   3.610   63.674000  0.074   
3    899  1993   7.882  0.011    -4.933    -4.958   5.411   58.403999  0.775   
4   1746  1993  24.496  0.000     7.723    -5.843  43.030  208.065000  0.719   
19  1686  1993   6.007  0.929     1.055     0.483   6.980   19.942678  0.086   
23   214  1993  17.609  7.213    -7.784    -9.292   8.150  132.936000  0.600   

       rd  threat         fv  rec1  rec2    mkt  rd_ratio        se  ad_ratio  \
0   0.305  1.7482  14.836461     0     0  2.042  0.084488 -0.056300  0.020499   
3   2.059  9.1024   6.930602     0     0  6.089  0.380521 -0.162903  0.143227   
4   0.372  1.0658   7.952849     0     0  9.994  0.008645  0.014166  0.016709   
19  0.197  1.0549   3.027747     0     0  1.558  0.028223 -0.018478  0.012321   
23  5.291  1.0538   7.814527     0     0  5.462  0.649202 -0.266398  0.073620   

   mkt_growth  
0              
3              
4              
19             
23

In [95]:
for company_id in data.id.unique():
    for year in data.year.unique():
        current = data.loc[data['id'] == company_id].loc[data.loc[data['id'] == company_id]['year'] == year].mv
        past = data.loc[data['id'] == company_id].loc[data.loc[data['id'] == company_id]['year'] == year-1].mv

        if current.shape[0] == 1 and past.shape[0] == 1:
            growth = (current.values[0]-past.values[0])/past.values[0]
            data.at[data.loc[data['id'] == company_id].loc[data.loc[data['id'] == company_id]['year'] == year-1].index[0], 'mkt_growth'] = growth

In [98]:
data.head()

id  year  assets   debt  earnings  cashflow   sales          mv     ad  \
0     42  1993   4.103  0.000    -1.748    -2.175   3.610   63.674000  0.074   
3    899  1993   7.882  0.011    -4.933    -4.958   5.411   58.403999  0.775   
4   1746  1993  24.496  0.000     7.723    -5.843  43.030  208.065000  0.719   
19  1686  1993   6.007  0.929     1.055     0.483   6.980   19.942678  0.086   
23   214  1993  17.609  7.213    -7.784    -9.292   8.150  132.936000  0.600   

       rd  threat         fv  rec1  rec2    mkt  rd_ratio        se  ad_ratio  \
0   0.305  1.7482  14.836461     0     0  2.042  0.084488 -0.056300  0.020499   
3   2.059  9.1024   6.930602     0     0  6.089  0.380521 -0.162903  0.143227   
4   0.372  1.0658   7.952849     0     0  9.994  0.008645  0.014166  0.016709   
19  0.197  1.0549   3.027747     0     0  1.558  0.028223 -0.018478  0.012321   
23  5.291  1.0538   7.814527     0     0  5.462  0.649202 -0.266398  0.073620   

   mkt_growth  
0              
3   -0.345079  
4   -0.002718  
19             
23